Modules/_decimal/_decimal.c

```diff
@@ -5857,7 +5876,7 @@ PyInit__decimal(void)
     mpd_free = PyMem_Free;
     mpd_setminalloc(_Py_DEC_MINALLOC);
```

Comment:
Yes. The warning occures here, which is caused by the multi-phase initialization of `decimal` module. @AlexWaygood 

This warning does not affect the module, because a static flag is used in `mpd_setminalloc` to ensure that `MPD_MINALLOC` can only be initialized once.

Maybe we can also add static flags to the `_decimal_exec` to suppress this warning. But I'm not sure if it makes sense to do so (we add a static variable just to tell the compiler that we know a warning will be raised here). :)

Reply:
I think it is reasonable. Warnings are disrupting.